src/main/java/net/minecraftforge/common/util/EnumHelper.java

```diff
@@ -141,6 +143,19 @@ public static EnumRarity addRarity(String name, TextFormatting color, String dis
     {
         return addEnum(EnumRarity.class, name, color, displayName);
     }
+    /**
+     * 
+     * @param name - The name the new enum constant should have
+     * @param textureLocation - The path to the texture for this armor type. Formatted as domain:path.
+     * @param hash - A short unique 3 letter name for the armor type 
+     * @param armorStrength - How much protection this armor type should give
+     * @return The created HorseArmorType enum constant
+     */
+    @Nullable 
+    public static HorseArmorType addHorseArmor(String name, String textureLocation, String hash, int armorStrength)
+    {
+        return addEnum(HorseArmorType.class, name, textureLocation, hash, armorStrength);
```

Comment:
This one is wrong... You don't need a `name` argument and you shouldn't have one. That's why the test is failing.